diff --git a/libavformat/tls_securetransport.c b/libavformat/tls_securetransport.c
index f52fc97..977b42b 100644
--- a/libavformat/tls_securetransport.c
+++ b/libavformat/tls_securetransport.c
@@ -37,6 +37,11 @@
 #include <Security/Security.h>
 #include <Security/SecureTransport.h>
 #include <CoreFoundation/CoreFoundation.h>
+#include <TargetConditionals.h>
+
+#ifndef TARGET_OS_MACCATALYST
+#define TARGET_OS_MACCATALYST 0
+#endif
 
 // We use a private API call here; it's good enough for WebKit.
 SecIdentityRef SecIdentityCreate(CFAllocatorRef allocator, SecCertificateRef certificate, SecKeyRef privateKey);
@@ -69,7 +74,7 @@ static int print_tls_error(URLContext *h, int ret)
 
 static int import_pem(URLContext *h, char *path, CFArrayRef *array)
 {
-#if !HAVE_SECITEMIMPORT
+#if !HAVE_SECITEMIMPORT || TARGET_OS_MACCATALYST
     return AVERROR_PATCHWELCOME;
 #else
     AVIOContext *s = NULL;
